modules/component/src/core/cd-aware/strategy.ts

```diff
@@ -0,0 +1,207 @@
+ * - VE/I - Options for ViewEngine / Ivy
+ * - mFC - `cdRef.markForCheck`
+ * - dC - `cdRef.detectChanges`
+ * - ɵMD - `ɵmarkDirty`
+ * - ɵDC - `ɵdetectChanges`
+ * - LV  - `LView`
+ * - C - `Component`
+ *
+ * | Name        | ZoneLess VE/I | Render Method VE/I  | Coalescing VE/I  |
+ * |-------------| --------------| ------------ ------ | ---------------- |
+ * | `native`    | ❌/❌         | mFC / mFC           | ❌               |
+ * | `global`    | ❌/✔️         | mFC  / ɵMD           | ❌              |
+ * | `local`     | ✔️/✔️          | dC / ɵDC            | ✔️ + C/ LV       |
+ * | `noop`      | ❌/❌         | no rendering        | ❌               |
+ *
+ */
+
+/**
+ * Native Strategy
+ * @description
+ *
+ * This strategy mirrors Angular's built-in `async` pipe.
+ * This means for every emitted value `ChangeDetectorRef#markForCheck` is called.
+ *
+ * | Name        | ZoneLess VE/I | Render Method VE/I  | Coalescing VE/I  |
+ * |-------------| --------------| ------------ ------ | ---------------- |
+ * | `native`    | ❌/❌         | mFC / mFC           | ❌               |
+ *
+ * @param config { StrategyFactoryConfig } - The values this strategy needs to get calculated.
+ * @return {CdStrategy<T>} - The calculated strategy
+ *
+ */
+export function createNativeStrategy<T>(
+  config: StrategyFactoryConfig
+): CdStrategy<T> {
+  return {
+    render: (): void => config.cdRef.markForCheck(),
+    behaviour: () => o => o,
+    name: 'native',
+  };
+}
+
+/**
+ * Noop Strategy
+ *
+ * This strategy is does nothing. It serves for debugging only
+ *
+ * | Name        | ZoneLess VE/I | Render Method VE/I  | Coalescing VE/I  |
+ * |-------------| --------------| ------------ ------ | ---------------- |
+ * | `noop`      | ❌/❌         | no rendering        | ❌               |
+ *
+ * @param config { StrategyFactoryConfig } - The values this strategy needs to get calculated.
+ * @return {CdStrategy<T>} - The calculated strategy
+ *
+ */
+export function createNoopStrategy<T>(): CdStrategy<T> {
+  return {
+    render: (): void => {},
+    behaviour: () => o => o,
+    name: 'noop',
+  };
+}
+
+/**
+ *
+ * Global Strategy
+ *
+ * This strategy is rendering the application root and
+ * all it's children that are on a path
+ * that is marked as dirty or has components with `ChangeDetectionStrategy.Default`.
+ *
+ * | Name        | ZoneLess VE/I | Render Method VE/I  | Coalescing VE/I  |
+ * |-------------| --------------| ------------ ------ | ---------------- |
+ * | `global`    | ❌/✔️         | mFC / ɵMD           | ❌               |
+ *
+ * @param config { StrategyFactoryConfig } - The values this strategy needs to get calculated.
+ * @return {CdStrategy<T>} - The calculated strategy
+ *
+ */
+export function createGlobalStrategy<T>(
+  config: StrategyFactoryConfig
+): CdStrategy<T> {
+  function render() {
+    if (!IS_VIEW_ENGINE_IVY) {
+      config.cdRef.markForCheck();
+    } else {
+      markDirty((config.cdRef as any).context);
+    }
+  }
```

Comment:
Nit, I find a negated if statement harder to read
```suggestion
    if (IS_VIEW_ENGINE_IVY) {
      markDirty((config.cdRef as any).context);
    } else {
      config.cdRef.markForCheck();
    }
  }
```

Reply:
fixed!